AirCasting/SDSync/ViewsFlow/SDSyncingAB/SDSyncViewModel.swift

```diff
@@ -63,13 +67,41 @@ class SDSyncViewModelDefault: SDSyncViewModel, ObservableObject {
                     }
                 }
             }, completion: { [weak self] result in
+                Log.info("[SD SYNC] Completed syncing with result: \(result)")
                 guard let self = self else { return }
-                if result {
+                switch result {
+                case .success():
+                    guard self.peripheral.state == .connected else {
+                        Log.info("[SD SYNC] Device disconnected. Attempting reconnect")
+                        self.reconnectWithAirbeamAndClearCard()
+                        return
+                    }
                     self.clearSDCard()
-                } else {
+                case .failure(let error):
                     DispatchQueue.main.async {
+                        switch error {
```

Comment:
would be nice to move this whole switch to a separate function (`self.alert = self.alertForError(error)`)